src/components/CompanySwitcher.tsx

```diff
@@ -43,73 +71,114 @@ export function CompanySwitcher() {
-                      <span>
-                        {record.company.settings.name || t('new_company')}
-                      </span>
-
-                      {state.currentIndex === index && <Check size={18} />}
-                    </div>
+    <>
+      <CompanyCreate
+        isModalOpen={isCompanyCreateModalOpened}
+        setIsModalOpen={setIsCompanyCreateModalOpened}
+      />
+
+      <CompanyEdit
+        isModalOpen={isCompanyEditModalOpened}
+        setIsModalOpen={setIsCompanyEditModalOpened}
+      />
+
+      <Menu as="div" className="relative inline-block text-left">
+        <div>
+          <Menu.Button className="flex items-center space-x-3 justify-center w-full rounded text-sm font-medium text-gray-700 border border-transparent">
+            <img className="w-8" src={logo} alt="Company logo" />
+            <span className="text-gray-900 text-sm">{companyName}</span>
+            <ChevronDown size={18} className="text-gray-900" />
+          </Menu.Button>
+        </div>
+
+        <Transition
+          as={Fragment}
+          enter="transition ease-out duration-100"
+          enterFrom="transform opacity-0 scale-95"
+          enterTo="transform opacity-100 scale-100"
+          leave="transition ease-in duration-75"
+          leaveFrom="transform opacity-100 scale-100"
+          leaveTo="transform opacity-0 scale-95"
+        >
+          <Menu.Items className="origin-top-right absolute left-0 mt-2 w-56 rounded shadow-lg bg-white ring-1 ring-black ring-opacity-5 divide-y divide-gray-100 focus:outline-none">
+            <div className="py-1">
+              <Menu.Item>
+                <DropdownElement>
+                  <p className="text-sm">{t('signed_in_as')}</p>
+                  <p className="text-sm font-medium text-gray-900 truncate">
+                    {user?.email}
+                  </p>
+                </DropdownElement>
+              </Menu.Item>
+            </div>
+
+            <div className="py-1">
+              {state?.api?.length >= 1 &&
+                state?.api?.map((record: any, index: number) => (
+                  <Menu.Item key={index}>
+                    <DropdownElement onClick={() => switchCompany(index)}>
+                      <div className="flex items-center space-x-3">
+                        <span>
+                          {record.company.settings.name || t('new_company')}
+                        </span>
+
+                        {state.currentIndex === index && <Check size={18} />}
+                      </div>
+                    </DropdownElement>
+                  </Menu.Item>
+                ))}
+            </div>
+            <div className="py-1">
+              {shouldShowAddCompany && canUserAddCompany && (
+                <Menu.Item>
+                  <DropdownElement
+                    className="flex items-center"
+                    onClick={() => setIsCompanyCreateModalOpened(true)}
+                  >
+                    {<PlusCircle />}
```

Comment:
No helper images. This is not the project's convention. We do buttons without images.

Reply:
> 

Sure, changed, I imported it from wrong package. But, always is okay to use react-icons when it is appropriate with UX?